src/core/core.layouts.js

```diff
@@ -199,38 +208,50 @@ function fitBoxes(boxes, chartArea, params, stacks) {
   return refit && fitBoxes(refitBoxes, chartArea, params, stacks) || changed;
 }
 
+function setBoxDims(box, left, top, width, height) {
+  box.top = top;
+  box.left = left;
+  box.right = left + width;
+  box.bottom = top + height;
+  box.width = width;
+  box.height = height;
+}
+
 function placeBoxes(boxes, chartArea, params, stacks) {
   const userPadding = params.padding;
-  let x = chartArea.x;
-  let y = chartArea.y;
-  let i, ilen, layout, box, stack;
+  let {x, y} = chartArea;
 
-  for (i = 0, ilen = boxes.length; i < ilen; ++i) {
-    layout = boxes[i];
-    box = layout.box;
-    stack = stacks[layout.pos + layout.stack] || {count: 1, placed: 0};
+  for (const layout of boxes) {
+    const box = layout.box;
+    const stack = stacks[layout.stack] || {count: 1, placed: 0, weight: 1};
     if (layout.horizontal) {
-      const width = chartArea.w / stack.count;
-      box.left = box.fullSize ? userPadding.left : chartArea.left + width * stack.placed;
-      box.right = box.fullSize ? params.outerWidth - userPadding.right : box.left + width;
-      box.top = y;
-      box.bottom = y + box.height;
-      box.width = box.right - box.left;
-      stack.placed++;
-      if (stack.placed === stack.count) {
-        y = box.bottom;
+      const width = chartArea.w / stack.weight * layout.stackWeight;
```

Comment:
If `stack.weight` or `layout.stackWeight` are 0 this will be a divide by 0. Perhaps factor that out into it's own variable `const weight = (stack.weight * layout.stackWeight) || 1`